skywalking-collector/skywalking-collector-cluster/src/main/java/com/a/eye/skywalking/collector/actor/AbstractASyncMemberProvider.java

```diff
@@ -0,0 +1,61 @@
+package com.a.eye.skywalking.collector.actor;
+
+import akka.actor.ActorRef;
+import com.a.eye.skywalking.collector.queue.DaemonThreadFactory;
+import com.a.eye.skywalking.collector.queue.MessageHolder;
+import com.lmax.disruptor.EventFactory;
+import com.lmax.disruptor.EventHandler;
+import com.lmax.disruptor.RingBuffer;
+import com.lmax.disruptor.dsl.Disruptor;
+
+import java.lang.reflect.Constructor;
+
+/**
+ * @author pengys5
+ */
+public abstract class AbstractASyncMemberProvider<T extends EventHandler> {
+
+    private RingBuffer<MessageHolder> ringBuffer;
+
+    public abstract Class memberClass();
+
+    public T createWorker(EventFactory eventFactory, ActorRef actorRef) throws Exception {
+        if (memberClass() == null) {
+            throw new IllegalArgumentException("cannot createInstance() with nothing obtained from memberClass()");
+        }
+
+        Constructor memberConstructor = memberClass().getDeclaredConstructor(new Class[]{ActorRef.class});
+        memberConstructor.setAccessible(true);
+        T member = (T) memberConstructor.newInstance(actorRef);
+
+        // Specify the size of the ring buffer, must be power of 2.
+        int bufferSize = 1024;
```

Comment:
Setting the `Disruptor` queue size as a const, which is fine, but the const should stay in `worker-module`.